drivers/serial/uart_stm32.c

```diff
@@ -456,8 +478,23 @@ static void uart_stm32_poll_out(const struct device *dev,
 	while (!LL_USART_IsActiveFlag_TXE(UartInstance)) {
 	}
 
-	/* do not allow system to suspend until transmission has completed */
-	pm_constraint_set(PM_STATE_SUSPEND_TO_IDLE);
+#ifdef CONFIG_PM
+	struct uart_stm32_data *data = DEV_DATA(dev);
+
+	if (!data->tx_poll_stream_on) {
+		data->tx_poll_stream_on = true;
+
+		/* Don't allow system to suspend until stream
+		 * transmission has completed
+		 */
+		uart_stm32_pm_constraint_set(dev);
+
```

Comment:
does it save some code with   `uart_stm32_pm_constraint_set(data)

Reply:
@FRASTM, I don't get this comment

Reply:
writing 
`static void uart_stm32_pm_constraint_set(const struct uart_stm32_data *data);`
instead of 
`static void uart_stm32_pm_constraint_set(const struct device *dev);`

Reply:
Both cases use a pointer, I'm not sure this would make a difference.
And then we'd need to extract `data` in each callers, which is not done yet.